File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/presentation/main/MainContract.java
package com.test.amaro.amarotest.presentation.main;

import com.test.amaro.amarotest.models.Product;
import com.test.amaro.amarotest.presentation.core.BaseMvpPresenter;
import com.test.amaro.amarotest.presentation.core.MvpView;
import java.util.List;

public interface MainContract {

    interface MainView extends MvpView {

        void reload();

        void onLoadStarted();

        void onLoadFinished();

        void onShowError(String error);

        void showProducts(List<Product> var1);
    }

    abstract class Presenter extends BaseMvpPresenter<MainView> {

        public abstract void loadProducts();
    }
}

File: AmaroTest/app/src/main/java/com/test/amaro/amarotest/main/MainContract.kt
package com.test.amaro.amarotest.main

import com.test.amaro.amarotest.core.BaseMvpPresenter
import com.test.amaro.amarotest.core.MvpView
import com.test.amaro.amarotest.models.Product

object MainContract {

    interface View : MvpView {

        fun retry()
        fun onLoadStarted()
        fun onLoadFinished()
        fun showProducts(products: List<Product>)
    }

    abstract class Presenter : BaseMvpPresenter<View>() {

        abstract fun getProducts()
    }
}